cfg/githubfieldmapper.go

```diff
@@ -0,0 +1,235 @@
+package cfg
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/andygrunwald/go-jira"
+	"github.com/coreos/issue-sync/lib/models"
+	"strings"
+	"time"
+)
+
+type FieldMapper interface {
+	MapFields(issue *models.ExtendedGithubIssue) jira.IssueFields
+
+	GetFieldValue(jIssue *jira.Issue, fieldKey FieldKey) (interface{}, error)
+
+	// getFieldIDs requests the metadata of every issue field in the JIRA
+	// project, and saves the IDs of the custom fields used by issue-sync.
+	GetFieldIDs(client jira.Client) (map[FieldKey]string, error)
+}
+
+type DefaultFieldMapper struct {
+	Config *Config
+}
+
+type JsonFieldMapper struct {
+	Config *Config
+}
+
+func (m DefaultFieldMapper) GetFieldValue(jIssue *jira.Issue, fieldKey FieldKey) (interface{}, error) {
+	switch fieldKey {
+	case GitHubID:
+		fallthrough
+	case GitHubNumber:
+		return jIssue.Fields.Unknowns.Int(m.Config.GetCompleteFieldKey(fieldKey))
+	default:
+		result, exists := jIssue.Fields.Unknowns.Value(m.Config.GetCompleteFieldKey(fieldKey))
+		if !exists {
+			return nil, errors.New("Field not found")
+		}
+		return result, nil
+	}
+}
+
+func (m DefaultFieldMapper) MapFields(issue *models.ExtendedGithubIssue) jira.IssueFields {
+	fields := jira.IssueFields{
+		Type: jira.IssueType{
+			Name: "Task", // TODO: Determine issue type
+		},
+		Project:     m.Config.GetProject(),
+		Summary:     issue.GetTitle(),
+		Description: issue.GetBody(),
+		Labels: 	 issue.CommitIds,
+		Unknowns:    map[string]interface{}{},
+	}
+
+	fields.Unknowns[m.Config.GetCompleteFieldKey(GitHubID)] = issue.GetID()
```

Comment:
I have been in this code a while, and still don't know why this is called "Unknowns".

Reply:
That is the Jira name for the field. I have no control over that.